lib/logstash/inputs/elasticsearch.rb

```diff
@@ -262,22 +268,41 @@ def run(output_queue)
   end
 
   private
-
+  JOB_NAME = "run query"
   def do_run(output_queue)
     # if configured to run a single slice, don't bother spinning up threads
-    return do_run_slice(output_queue) if @slices.nil? || @slices <= 1
+    if @slices.nil? || @slices <= 1
+      return retryable(JOB_NAME) do
+        do_run_slice(output_queue)
+      end
+    end
 
     logger.warn("managed slices for query is very large (#{@slices}); consider reducing") if @slices > 8
 
     pipeline_id = execution_context&.pipeline_id || 'main'
     @slices.times.map do |slice_id|
       Thread.new do
         LogStash::Util::set_thread_name("[#{pipeline_id}]|input|elasticsearch|slice_#{slice_id}")
-        do_run_slice(output_queue, slice_id)
+        retryable(JOB_NAME) do
+          do_run_slice(output_queue, slice_id)
+        end
       end
     end.map(&:join)
   end
 
+  def retryable(job_name, &block)
+    begin
+      stud_try = ::LogStash::Helpers::LoggableTry.new(logger, job_name)
+      stud_try.try((@retries + 1).times) {
+        block.call
+      }
```

Comment:
No need to create a new closure/binding. We can pass the block through.

```suggestion
      stud_try.try((@retries + 1).times, &block)
```

It used to be that `yield` was a bit faster than `block.call`, so perhaps this would be a more readable variant:

```suggestion
      stud_try.try((@retries + 1).times) { yield }
```